Update styles of regional constraint track

nawatts opened the pull request:
* Color regions based on their significance and observed/expected missense.
* Show observed/expected missense in region.
* Add help button.
* When hovering the mouse over a region, show a tooltip with observed/expected missense and significance. 
* For consistency with other Track components, move RegionalConstraintTrack into its own package instead of defining it inline in projects/gnomad/RegionViewer.

## Before
<img width="1110" alt="screen shot 2018-06-20 at 10 12 52 am" src="https://user-images.githubusercontent.com/1156625/41663997-b93f1d32-7472-11e8-9e72-07caf8488a31.png">

## After
<img width="1087" alt="screen shot 2018-06-20 at 10 13 34 am" src="https://user-images.githubusercontent.com/1156625/41663998-b9502f50-7472-11e8-8783-d6598486b18f.png">
<img width="990" alt="screen shot 2018-06-20 at 10 12 24 am" src="https://user-images.githubusercontent.com/1156625/41663999-b95db314-7472-11e8-8786-c7894e7922ff.png">

anneodonnell commented:
I think this looks pretty nice. We probably need a brief description of it in the FAQ as I'm sure we will get questions about what the numbers mean. 

I wonder if the regional constraint table should have some kind of O/E mention also to give people a clue about these numbers.

nawatts commented:
We have some documentation for regional constraint here: https://github.com/macarthur-lab/gnomad-docs/blob/master/docs/regional-constraint.md

We can add a help icon/popup to the track to show this.

ksamocha commented:
To @anneodonnell's point, you might want to write "O/E" instead of gamma. We only used gamma in the paper for ease.

ksamocha commented:
I'll note here the point I brought up at the beginning of the call today: there are two separate pieces of information for all gene/region -- obs/exp and signficance (chisq_diff_null). Currently, gnomAD is coloring by obs/exp to highlight extremely depleted regions. DECIPHER decided to only color regions by their obs/exp value, but only if those regions were significantly different from expectation.

I don't think there is a correct way to do this. There are 143 regions that have obs/exp < 0.6 and a chi-squared value that we wouldn't consider significant (< 10.8).
```
> awk '{ if($9<0.6 && $10<10.8) print }' genes_regional_data_cleaned_chisq_0.0_metrics_bpadded2_2016_10_05.txt | wc -l
     143
```

Something to think about before fully approving this. Otherwise, I think it looks pretty good.

nawatts commented:
Changed gamma to O/E and added a help button.

<img width="1486" alt="screen shot 2018-06-07 at 11 26 06 am" src="https://user-images.githubusercontent.com/1156625/41109715-aa334cfa-6a45-11e8-863c-9b1f4be7ceb3.png">

nawatts commented:
Based on feedback from today's meeting, the remaining todos are:

* [ ] Change label to "Regional missense constraint" (this also affects help text)
* [ ] Remove O/E labels from regions and instead show them in a tooltip on hover
* [ ] Color regions based on significance instead of observed/expected variation

nawatts commented:
Updated the PR description with current screenshots.

nawatts commented on packages/api/package.json:
```diff
@@ -25,6 +25,7 @@
   "license": "ISC",
   "dependencies": {
     "compression": "^1.7.1",
+    "core-js": "^2.5.0",
```
This was necessary to solve some version mismatch problem in `babel-polyfill` after updating dependencies.

mattsolo1 commented on packages/ui/src/tooltip/withTooltip.js:
```diff
@@ -0,0 +1,94 @@
+}
+
+
+export function withTooltip(Component) {
```
Love this!

mattsolo1 commented on packages/track-regional-constraint/src/RegionalConstraintTrack.js:
```diff
@@ -0,0 +1,149 @@
+}
+
+
+const WithRegionTooltip = withTooltip(({ region }) => (
```
This tooltip component is beautiful
